client/app/components/CommentForm.jsx

```diff
@@ -10,31 +10,27 @@ import ReactCSSTransitionGroup from 'react/lib/ReactCSSTransitionGroup';
 const emptyComment = { author: '', text: '' };
 const textPlaceholder = 'Say something using markdown...';
 
-const CommentForm = React.createClass({
-  displayName: 'CommentForm',
+class CommentForm extends React.Component {
+  state = {
+    formMode: 0,
+    comment:  emptyComment
+  };
 
-  propTypes: {
+  static displayName = 'CommentForm';
+
+  static propTypes = {
     ajaxSending: PropTypes.bool.isRequired,
     actions: PropTypes.object.isRequired,
     error: PropTypes.any,
-  },
-
-  getInitialState() {
-    return {
-      formMode: 0,
-      comment: emptyComment,
-    };
-  },
+  };
 
-  handleSelect(selectedKey) {
+  handleSelect = (selectedKey) => {
```

Comment:
@rstudner I am new to React/ES6, I am wondering why is this pattern better that the original one?